File: API/Data/Entities/Product.cs
using System;
using System.Collections.Generic;

namespace API.Data.Entities
{
    public partial class Product
    {
        public Product()
        {
            ProductColors = new HashSet<ProductColor>();
            ProductImages = new HashSet<ProductImage>();
            UserLikes = new HashSet<UserLike>();
        }

        public int Id { get; set; }
        public string Name { get; set; } = null!;
        public int Price { get; set; }
        public string? Description { get; set; }
        public int Discount { get; set; }
        public string Slug { get; set; } = null!;
        public DateTime? CreateAt { get; set; }
        public DateTime? UpdateAt { get; set; }
        public int? CategoryId { get; set; }
        public bool IsVisible { get; set; }
        public string? ModifiedBy { get; set; }

        public virtual Category? Category { get; set; }
        public virtual ICollection<ProductColor> ProductColors { get; set; }
        public virtual ICollection<ProductImage> ProductImages { get; set; }
        public virtual ICollection<UserLike> UserLikes { get; set; }
    }
}

File: Client/src/app/shared/models/product.ts
import { Category } from "./category"
import { Color } from "./color"

export interface Product {
    id: string
    name: string
    price: number
    mainPhoto: string
    subPhoto: string
    productColors: Color[]
    discount: number
    isVisible: boolean
    slug: string
    category: string
    createAt: string
    isLikedByCurrentUser: boolean
  }

  export interface ProductDetail {
    id: string
    name: string
    price: number
    description: string
    discount: number
    slug: string
    category: Category
    productImages: ProductImage[]
    productVariants: ProductVariant[]
  }

  export interface ProductImage {
    id: number
    imageUrl: string
    isMain: boolean
    isSub: boolean
  } 
  
  export interface ProductVariant {
    id: number
    quantity: number
    color: string
    colorCode: string
    size: string
  }

  export interface CreateProduct {
    name: string
    price: number
    description: string
    discount: number
    categoryId: number
    productColors: number[]
    productSizes: number[]
    mainImage: File
    subImage: File
    productImages: File[]
  }

  export interface UpdateProduct {
  name: string
  price: number
  description: string
  discount: number
  categoryId: number
  productColors: number[]
  productSizes: number[]
}